filecoin/retriever.go

```diff
@@ -586,6 +600,19 @@ func (retriever *Retriever) OnRetrievalEvent(event rep.RetrievalEvent) {
 			event.StorageProviderId(),
 			ret.QueryResponse(),
 		)
+		if ret.QueryResponse().Status == retrievalmarket.QueryResponseError {
```

Comment:
🤦 so it's a "successful error" case; whatever

Reply:
I mean, we don't really have HTTP response codes, so yea.

The other stuff is Libp2p errors, not response errors.